scripts/publish_prod.sh

```diff
@@ -122,6 +122,10 @@ echo "Publishing to PyPI"
 echo "Please enter the token (password) for datadog-cdk-constructs-v2 PyPI"
 python3 -m twine upload -u __token__ ./dist/python/* 
 
+echo "Publishing the Go package to datadog-cdk-constructs-go GitHub repo"
+read -p "Please enter your GitHub username: " GITHUB_TOKEN
```

Comment:
Is the `username` the token?

Reply:
Yes.
### Example 1
If I run the command without providing GITHUB_TOKEN, then it will say GITHUB_TOKEN is required.
```
yiming.luo@COMP-J7WRL6274T datadog-cdk-constructs % unset GITHUB_TOKEN                                                                                        
yiming.luo@COMP-J7WRL6274T datadog-cdk-constructs % echo $GITHUB_TOKEN

yiming.luo@COMP-J7WRL6274T datadog-cdk-constructs % npx -p publib@latest publib-golang
Detected modules:
 - /Users/yiming.luo/go/src/github.com/DataDog/datadog-cdk-constructs/dist/go/ddcdkconstruct/go.mod
/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/help/git.js:43
            throw new Error('GITHUB_TOKEN env variable is required when GITHUB_USE_SSH env variable is not used');
            ^

Error: GITHUB_TOKEN env variable is required when GITHUB_USE_SSH env variable is not used
    at Object.clone (/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/help/git.js:43:19)
    at GoReleaser.release (/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/targets/go.js:72:13)
    at Object.<anonymous> (/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/bin/publib-golang.js:37:10)
    at Module._compile (node:internal/modules/cjs/loader:1467:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1551:10)
    at Module.load (node:internal/modules/cjs/loader:1282:32)
    at Module._load (node:internal/modules/cjs/loader:1098:12)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:215:24)
    at Module.require (node:internal/modules/cjs/loader:1304:12)

Node.js v22.4.1
yiming.luo@COMP-J7WRL6274T datadog-cdk-constructs % 
```

### Example 2
If I run the script and provide a wrong GITHUB_TOKEN (`yiming_luo_not_exist`), then it will ask for password for `https://yiming_luo_not_exist@github.com`
```
yiming.luo@COMP-J7WRL6274T datadog-cdk-constructs % GIT_BRANCH=yiming.luo/go-publish-test GITHUB_TOKEN=yiming_luo_not_exist npx -p publib@latest publib-golang
Detected modules:
 - /Users/yiming.luo/go/src/github.com/DataDog/datadog-cdk-constructs/dist/go/ddcdkconstruct/go.mod
Cloning into '/var/folders/25/9yj27sqs2b50bn5m48th2spm0000gp/T/mieopw/repo'...
remote: Enumerating objects: 40, done.
remote: Counting objects: 100% (40/40), done.
remote: Compressing objects: 100% (29/29), done.
remote: Total 40 (delta 6), reused 37 (delta 4), pack-reused 0
Receiving objects: 100% (40/40), 2.75 MiB | 11.48 MiB/s, done.
Resolving deltas: 100% (6/6), done.
[origin/yiming.luo/go-publish-test] fix: Delete unused files at root directory
branch 'yiming.luo/go-publish-test' set up to track 'origin/yiming.luo/go-publish-test'.
Switched to a new branch 'yiming.luo/go-publish-test'
:100644 100644 7f266e009195ab8d339619b6d9acd7348458a4d0 5f3cc374d066ec585f480ee6dd9d517e37d9b3b2 M	ddcdkconstruct/README.md
:100644 100644 6c5e8a8fa55093a514ef800def3b5bc30e5940f5 2895625351bcf0e9be170dfdb0367bf970697fe4 M	ddcdkconstruct/jsii/datadog-cdk-constructs-v2-1.13.0.tgz
:000000 100644 0000000000000000000000000000000000000000 d9c6852f7b82f1fe3a618b20734d90726102c890 A	ddcdkconstruct/jsii/datadog-cdk-constructs-v2-1.15.0.tgz
[yiming.luo/go-publish-test f95b7e7] chore(release): v1.13.0
 3 files changed, 74 insertions(+), 42 deletions(-)
 create mode 100644 ddcdkconstruct/jsii/datadog-cdk-constructs-v2-1.15.0.tgz
Password for 'https://yiming_luo_not_exist@github.com': 
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/DataDog/datadog-cdk-constructs-go.git/'
/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/help/shell.js:56
        throw new Error(message);
        ^

Error: 
    Command failed: git push origin ddcdkconstruct/v1.13.0
.
      Output: undefined
      Error: undefined
    at Object.run (/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/help/shell.js:56:15)
    at Object.push (/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/help/git.js:153:11)
    at /Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/targets/go.js:114:35
    at Array.forEach (<anonymous>)
    at GoReleaser.doRelease (/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/targets/go.js:114:18)
    at GoReleaser.release (/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/targets/go.js:76:25)
    at Object.<anonymous> (/Users/yiming.luo/.npm/_npx/09fe13222ade61c7/node_modules/publib/lib/bin/publib-golang.js:37:10)
    at Module._compile (node:internal/modules/cjs/loader:1467:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1551:10)
    at Module.load (node:internal/modules/cjs/loader:1282:32)

Node.js v22.4.1
yiming.luo@COMP-J7WRL6274T datadog-cdk-constructs %
```

### Example 3
See the example in "Testing Guidelines ". Only when I use `GITHUB_TOKEN=lym953` will the command run properly.